core/src/main/scala/caliban/rendering/DocumentRenderer.scala

```diff
@@ -322,6 +336,64 @@ object DocumentRenderer extends Renderer[Document] {
         }
     }
 
+  private trait SchemaOps {
+    def unsafeRenderImpl(
+      theType: String,
+      description: Option[String],
+      directives: List[Directive],
+      query: Option[String],
+      mutation: Option[String],
+      subscription: Option[String],
+      indent: Option[Int],
+      writer: StringBuilder
+    ) = {
+      def renderOp(name: String, op: Option[String]): Unit =
+        op.foreach { o =>
+          pad(increment(indent), writer)
+          writer append name
+          writer append ':'
+          spaceOrEmpty(indent, writer)
+          writer append o
+          newlineOrEmpty(indent, writer)
+        }
+      newlineOrEmpty(indent, writer)
+      descriptionRenderer.unsafeRender(description, indent, writer)
+      writer append theType
+      spaceOrEmpty(indent, writer)
+      directivesRenderer.unsafeRender(directives, indent, writer)
+      val hasTypes                                         = query.nonEmpty || mutation.nonEmpty || subscription.nonEmpty
+      if (hasTypes) {
+        writer append '{'
+        newlineOrEmpty(indent, writer)
+        renderOp("query", query)
+        renderOp("mutation", mutation)
+        renderOp("subscription", subscription)
+        writer append '}'
+      }
+      newlineOrEmpty(indent, writer)
+    }
+  }
+  private lazy val schemaExtensionRenderer: Renderer[TypeSystemExtension.SchemaExtension] =
+    new Renderer[TypeSystemExtension.SchemaExtension] with SchemaOps {
+      override protected[caliban] def unsafeRender(
+        extension: TypeSystemExtension.SchemaExtension,
+        indent: Option[Int],
+        write: StringBuilder
+      ): Unit = {
+        import extension._
+        unsafeRenderImpl("extend schema", None, directives, query, mutation, subscription, indent, write)
+      }
+    }
+  // TODO: The implementation of schemaRenderer is wrong. Two schema definations is invalid. The current implementation
+  // treats it as an extension. The correct implementation should be the follwoing one but I'm keeping the old one for
+  // backward compatibility for now.
+  //  private lazy val schemaRenderer: Renderer[SchemaDefinition] = new Renderer[SchemaDefinition] with SchemaOps {
+  //    override def unsafeRender(definition: SchemaDefinition, indent: Option[Int], writer: StringBuilder): Unit = {
+  //      import definition._
+  //      unsafeRenderImpl("schema", description, directives, query, mutation, subscription, indent, writer)
+  //    }
+  //  }
```

Comment:
Im ok with this change generally, as long as the current behavior (for partial schemas such as in federation) still works. The issue with the change as is, is that when going from GraphQL to renderer schema we don't do any extension types. I'd suggest that we instead then update the GraphQL#toDocument method to output the schema as an extension type if it's subgraph variant.